src/Pages/StockDetail/StockDetail.tsx

```diff
@@ -1,36 +1,48 @@
-import React from 'react';
-
-import { useParams } from 'react-router-dom';
+import axios from 'axios';
+import { useParams } from 'react-router-dom'
 import { useQuery } from 'react-query';
-import { Stock } from './components/Stock/Stock';
-import { useStockUi } from './hooks/useStockUI';
+import { Stock } from '../../state/watchlist/watchlistSlice';
 
-function StockDetail() {
-  //я когда раньше использовал useParams, он мне возразал значение строкой, а сейчас приходит объект
-  const { data, loading, error } = useStockUi();
 
-  if (loading) {
-    return <h3>Loading...</h3>;
+type StockDetailUrlParams = string
+
+function StockDetail() {
+  const stock_id = useParams<StockDetailUrlParams>()
+  async function fetchStockInfo(){
+    const { data } = await axios.get(`/api/stock/${stock_id.id}`)
+    return data
+  }
+  const { data, isLoading, isError } = useQuery<Stock>({ queryKey: ['stockInfo'], queryFn: fetchStockInfo })
+  if(isLoading){
+    return <h3>Loading...</h3>
   }
 
-  if (error) {
+  if (isError) {
     return <h3>Mistake with data fetching</h3>;
   }
 
   if (!data) {
     return <h3>No data</h3>;
   }
 
+  const { image_url, stock_name, stock_price, id, index, isin }: Stock = data;
```

Comment:
А ты же передаешь вроде как тип data вот тут: https://github.com/Alexander-Mentoring/ISIN_APP_TS/pull/21/files#diff-27897b11bf13163c8c2a0eea7bd902476ad3620ece49f876a7998202b71de1f9R15

Тогда дополнительно определять тип Stock не нужно

```suggestion
  const { image_url, stock_name, stock_price, id, index, isin } = data;
```